datarepo-clienttests/src/main/java/runner/TestRunner.java

```diff
@@ -178,16 +199,16 @@ void executeTestConfigurationNoGuaranteedCleanup() throws Exception {
 
     // wait until all threads either finish or time out
     logger.info("Test Scripts: Waiting until all threads either finish or time out");
-    for (int ctr = 0; ctr < scripts.size(); ctr++) {
-      TestScriptSpecification testScriptSpecification = config.testScripts.get(ctr);
-      ThreadPoolExecutor threadPool = threadPools.get(ctr);
+    Iterator it = threadPools.entrySet().iterator();
+    while (it.hasNext()) {
+      HashMap.Entry threadPoolEntry = (HashMap.Entry) it.next();
+      ThreadPoolExecutor threadPool = (ThreadPoolExecutor) threadPoolEntry.getValue();
+      SpecificationInterface script = (SpecificationInterface) threadPoolEntry.getKey();
 
       threadPool.shutdown();
-      long totalTerminationTime =
-          testScriptSpecification.expectedTimeForEach * testScriptSpecification.totalNumberToRun;
+      long totalTerminationTime = script.expectedTimeForEach * script.totalNumberToRun;
       boolean terminatedByItself =
-          threadPool.awaitTermination(
-              totalTerminationTime, testScriptSpecification.expectedTimeForEachUnitObj);
+          threadPool.awaitTermination(totalTerminationTime, script.expectedTimeForEachUnitObj);
```

Comment:
I don't think we need to wait for the failure pattern to finish -- there's no point to letting them continue if all the user journeys are done is there? We can just wait for the user journeys to finish/timeout and then kill the failure pattern script.

Reply:
Ah, makes sense - I have changed it so that once all of the test scripts have finished, we force shutdown the disruption script. This means that the disruption script thread pool is tracked separately from the test script thread pools.